src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmEvaluationAsyncResult.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#region Assembly Microsoft.VisualStudio.Debugger.Engine, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
+// D:\Roslyn\Main\Open\Binaries\Debug\Microsoft.VisualStudio.Debugger.Engine.dll
+#endregion
+
+using System;
+
+namespace Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation
+{
+    public struct DkmEvaluationAsyncResult
+    {
+        public DkmEvaluationAsyncResult(DkmEvaluationResult Result)
+        {
+            throw new NotImplementedException();
+        }
+
+        public int ErrorCode { get { throw new NotImplementedException(); } }
+
+        public DkmEvaluationResult Result { get { throw new NotImplementedException(); } }
+
+        public static DkmEvaluationAsyncResult CreateErrorResult(Exception exception)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
Clearly not... :)

Looks like I can change the test helper to call through the public entry point (covers the normal constructor case).  I'm not sure the CreateErrorResult case though...  It appears that it will only ever be called if the constructor throws.  I can write some sort of helper to make that happen, but that seems like a lot of overhead to test a narrow case.  I can certainly put a null-check/throw in the constructor if "Result == null" (most likely reason the constructor might throw).

Thoughts?